### PR TITLE
Feature/add cmek support 114

### LINKED ISSUE #114: Add CMEK support for state file GCS bucket
Add capability to encrypt state file bucket using CMEK.

### PR DESCRIPTION
**Pull request by imrannayer**
Fixed #114

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: main.tf [hunk 1]
```diff
@@ -80,6 +80,40 @@ resource "google_service_account" "org_terraform" {
...
+  }
+
+}
+
+resource "google_kms_crypto_key_iam_member" "gcs_key_iam" {
+  count         = var.encrypt_gcs_bucket_tfstate ? 1 : 0
+  crypto_key_id = google_kms_crypto_key.gcs_key[0].id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+  member        = "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"
+}
```
Comment (bharathkkb) [inline main.tf:116]: Could we use https://github.com/terraform-google-modules/terraform-google-kms?
```tf
module "kms" {
  source  = "terraform-google-modules/kms/google"
  version = "~> 1.2"

  project_id         = module.seed_project.project_id
  location           = var.default_region
  keyring            = "${var.project_prefix}-keyring"
  keys               = [ "${var.project_prefix}-key"]
  key_rotation_period = var.key_rotation_period
  key_protection_level = var. key_protection_level
  set_decrypters_for  = ["${var.project_prefix}-key"]
  set_encrypters_for = ["${var.project_prefix}-key"]
  decrypters = [
   "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}",
  ]
  encrypters = [
   "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}",
  ]
}
```

Comment (imrannayer) [reply main.tf:116]: Definitely. Leme update.

File: main.tf [hunk 2]
```diff
@@ -91,6 +116,14 @@ resource "google_storage_bucket" "org_terraform_state" {
   versioning {
     enabled = true
   }
+
+  dynamic "encryption" {
+    for_each = var.encrypt_gcs_bucket_tfstate ? ["encryption"] : []
+    content {
+      # default_kms_key_name = google_kms_crypto_key.gcs_key[0].id
```
Comment (bharathkkb) [inline main.tf:123]: nit: comment

**Review by bharathkkb — APPROVED** [review]
LGTM
/cc @rjerrems

File: main.tf (hunk 2 above)
Comment (imrannayer) [reply main.tf:123]: My bad. Leme fix it.

File: variables.tf [hunk 3]
```diff
@@ -164,3 +165,25 @@ variable "random_suffix" {
...
+  description = "The protection level to use when creating a version based on this template. Default value: \"SOFTWARE\" Possible values: [\"SOFTWARE\", \"HSM\"]"
+  default     = "SOFTWARE"
+}
+
+variable "key_rotation_period" {
+  type    = string
+  default = null
+}
+
+variable "prevent_destroy" {
```
Comment (rjerrems) [inline variables.tf:186]: nit: If this only applies to KMS, it should probably be prefixed with `kms_`

**Comment by rjerrems** [comment]
I don't seem to have the option to add my review on this repo @bharathkkb - PR LGTM, just one minor nit on variable naming.

**Review by rjerrems — DISMISSED** [review]

**Comment by rjerrems** [comment]
@imrannayer can you also please rebase from master as your other change has been merged.

File: variables.tf (hunk 3 above)
Comment (imrannayer) [reply variables.tf]: done

**Review by bharathkkb — COMMENTED** [review]
@imrannayer looks like there are changes from https://github.com/terraform-google-modules/terraform-google-bootstrap/pull/116 in this PR. Could you rebase?

**Review by rjerrems — APPROVED** [review]

**Comment by rjerrems** [comment]
Thanks @imrannayer - I have created a follow up issue here https://github.com/terraform-google-modules/terraform-google-bootstrap/issues/119 for add an example and tests

**Pull request merged**